File: frontend/src/types/domain.ts
export type DomainStatus =
  | 'active'
  | 'inactive'
  | 'error'
  | 'pending'
  | 'valid'
  | 'junk'
  | 'dead'
  | 'alive'

export type Domain = {
  id: string;
  url: string;
  domain_type: 'website' | 'api' | 'other';
  status: DomainStatus;
  health?: DomainStatus;
  country: string
  auth_status: boolean;
  last_checked: string; // ISO 8601 string
  response_time: number; // ms
  error_message: string | null;
  created_at: string; // ISO 8601 string
  updated_at: string; // ISO 8601 string
};

export type AddDomainInput = {
  url: string;
  domain_type: 'website' | 'api' | 'other';
};

export type EditDomainInput = {
  url: string;
  domain_type: 'website' | 'api' | 'other';
  status: DomainStatus;
  auth_status: boolean;
  response_time: number;
  error_message: string | null;
};

File: backend/schemas/domain.py
from datetime import datetime
from enum import Enum
from uuid import UUID

from pydantic import BaseModel


class DomainStatus(str, Enum):
    NONE = "none"
    CHECKED = "checked"
    DEAD = "dead"


class DomainBase(BaseModel):
    url: str
    domain_type: str | None = None
    country: str | None = None


class DomainCreate(DomainBase):
    pass


class DomainUpdate(BaseModel):
    url: str | None = None
    domain_type: str | None = None
    status: DomainStatus | None = None
    auth_status: bool | None = None
    last_checked: datetime | None = None
    response_time: float | None = None
    error_message: str | None = None
    country: str | None = None


class Domain(DomainBase):
    id: UUID
    status: DomainStatus = DomainStatus.NONE
    auth_status: bool = False
    last_checked: datetime | None = None
    response_time: float | None = None
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime | None = None
